numpy/lib/shape_base.py

```diff
@@ -422,18 +421,23 @@ def array_split(ary, indices_or_sections, axis=0):
         Ntotal = len(ary)
     try:
         # handle scalar case.
-        Nsections = len(indices_or_sections) + 1
-        div_points = [0] + list(indices_or_sections) + [Ntotal]
+        Nsections = len(chunks) + 1
+        div_points = [0] + list(chunks) + [Ntotal]
     except TypeError:
-        # indices_or_sections is a scalar, not an array.
-        Nsections = int(indices_or_sections)
-        if Nsections <= 0:
-            raise ValueError('number sections must be larger than 0.')
-        Neach_section, extras = divmod(Ntotal, Nsections)
-        section_sizes = ([0] +
-                         extras * [Neach_section+1] +
-                         (Nsections-extras) * [Neach_section])
-        div_points = _nx.array(section_sizes).cumsum()
+        # chunks is not iterable
+        try:
+            Nsections = int(chunks)
+        except TypeError:
+            # chunks cannot be converted to int
+            Nsections = Ntotal
+        finally:
```

Comment:
This should not be a `finally` - if `int(chunks)` throws `ValueError`, then `Nsections` would not even be set, and this would give a `NameError`